### PR TITLE
Changed keyboardtype and lockscreen behavior 

### PR DESCRIPTION
**Pull request by Drudoo**
I have changed the keyboardtype to UIKeyboardAppearanceAlert to get the transparent keyboard. 
Screenshot: http://cl.ly/3B032f042J1v230O1s1b

Changed Lockscreen from using alertTableViewDataSource to alertTableViewDataSourceEditable this makes the tableview editable and then I added didSelectRowAtIndexPath so we can go from our Lockscreen directly to our app.

Screenshot: http://cl.ly/0i3r1H30251u3L2T300h

There is a lot of Commits but it is just a few things, which have been tested back and forth =)

### CODE REVIEW

**Comment by timnovinger** [comment]
@Drudoo - I will test this out tonight. I just built it and installed it onto my phone.

I like the idea of the keyboard style change, but I'm very hesitant about making the lockscreen notification table editable as it introduces issues.

**Comment by neelo** [comment]
This is great news
If only you can get permission off Peter to upload your betas to a repo for the people who cannot build this

File: MNAlertViewController.m [hunk 1]
```diff
@@ -118,7 +118,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 	dateText.textColor = [UIColor colorWithRed:1.000 green:1.000 blue:1.000 alpha:0.5];
 	dateText.backgroundColor = [UIColor clearColor];
 	[dateText setAlpha:0.0];
-	
+
```
Comment (timnovinger) [inline MNAlertViewController.m:121]: What's with all of these strange whitespace commits?

**Comment by timnovinger** [comment]
Unfortunately this still needs some work before it's perfect. Here are the issues that I've noticed:
- If a notification is deleted from the lockscreen list the "missed notifications" count is not updated (or removed if it's the last or only notification in the list)
- If the phone is setup with a passcode this does not have the desired effect of bypassing the lockscreen and going directly to the application. Instead, it removes the lockscreen notification list and the user is left at the lockscreen not knowing what just happened. This is **the** main reason that Peter and I decided against this functionality on the lockscreen – it's supposed to be locked and non-interactive on purpose for security reasons.

In the meantime I'm going to add in the keyboard style change since it's just one line (unfortunately I'll have to commit this myself since I can't pull _just_ that out of your pull request).

Thanks for the help, and hopefully you can solve the issues with this feature!

File: MNAlertViewController.m [hunk 2]
```diff
@@ -184,9 +184,11 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 		charactersTyped.textColor = [UIColor colorWithRed:1.000 green:1.000 blue:1.000 alpha:0.5];
 		charactersTyped.backgroundColor = [UIColor clearColor];
 		[charactersTyped setAlpha:0.0];
-		
+
 		textBox = [[BCZeroEdgeTextView alloc] initWithFrame:CGRectMake(17.0, 155.0, 285.0, 36.0)];
 		[textBox setDelegate:self];
+//tested and works. Ready to be implemented.
+                textBox.keyboardAppearance = UIKeyboardAppearanceAlert;
```
Comment (timnovinger) [inline MNAlertViewController.m:191]: This is a great idea.

**Comment by BDudely** [comment]
When I look at all this code i just see beautiful, beautiful, updates;)

**Comment by ghost** [comment]
not to be pushy or anything, but does this also fixes the notification bug, when you get a popup and reply to a message to clear the notification badge on the SMS app?

**Comment by timnovinger** [comment]
Nope @unknownsolo, that is a completely separate issue altogether.

## 

## Tim Novinger

Twitter:  @timnovinger
Skype: tim.novinger
Mobile: 260.918.4937

## Web:  timnovinger.com

Sent with Sparrow (http://www.sparrowmailapp.com)

**Comment by Drudoo** [comment]
Thanks for the comments. 
After reading Tim's and Peter's comment, on issue #45, I will have to agree with both of you. I can see the problem with the Lockscreen and the way iOS is working. In this stage of development I can see that we have bigger issues and I will let the Lockscreen rest for a bit. I think the Lockscreen view needs a bit of a graphic change. The TableView should not be higher than the rows needed to display the notifications, that counts for the Dashboard too, so we don't have the transparent black background all over the screen. 
As for new modifications I think the Dashboard need to me fully working before the Lockscreen goes into heavy development. There is a big problem with dismissing the screen when not using the Switcher. When the Dashboard is shown at the Switcher there is a problem with Music Control Pro. 

As for the strange space whitespace commits i have no idea. Github shouldn't report those :/ 
I will try to make smaller pull request next time, so you can merge it ;)

The next thing i will focus on is the badge clearing, which is REALLY annoying, and better functions for the dashboard.

**Comment by timnovinger** [comment]
@drudoo Sorry we couldn't merge this code in, but your help is very appreciated. Keep up the good work dude!

**Pull request closed without merging**